HugeCTR/include/hps/hash_map_backend_detail.hpp

```diff
@@ -55,7 +56,9 @@ namespace HugeCTR {
       const Payload& payload{it->second};                           \
                                                                     \
       /* Stash pointer and reference in map. */                     \
+      std::unique_lock lock(part.read_write_lck);                   \
```

Comment:
@yingcanw  What worries me about this is mostly the strain under heavy load. We would need to measure if it doesn't kill insert speed. The new hashmap will fix this one.